feature/rds/auth/builder.go

```diff
@@ -0,0 +1,123 @@
+	creds    aws.CredentialsProvider
+
+	connectFormat ConnectionFormat
+	params        url.Values
+}
+
+// NewConnectionStringBuilder will return an ConnectionStringBuilder
+func NewConnectionStringBuilder(endpoint, region, dbUser, dbName string, creds aws.CredentialsProvider) ConnectionStringBuilder {
+	return ConnectionStringBuilder{
+		dbName:   dbName,
+		endpoint: endpoint,
+		region:   region,
+		user:     dbUser,
+		creds:    creds,
+	}
+}
+
+// WithEndpoint will return a builder with the given endpoint
+func (b ConnectionStringBuilder) WithEndpoint(endpoint string) ConnectionStringBuilder {
+	b.endpoint = endpoint
+	return b
+}
+
+// WithRegion will return a builder with the given region
+func (b ConnectionStringBuilder) WithRegion(region string) ConnectionStringBuilder {
+	b.region = region
+	return b
+}
+
+// WithUser will return a builder with the given user
+func (b ConnectionStringBuilder) WithUser(user string) ConnectionStringBuilder {
+	b.user = user
+	return b
+}
+
+// WithDBName will return a builder with the given database name
+func (b ConnectionStringBuilder) WithDBName(dbName string) ConnectionStringBuilder {
+	b.dbName = dbName
+	return b
+}
+
+// WithParams will return a builder with the given params. The parameters
+// will be included in the connection query string
+//
+//	Example:
+//	v := url.Values{}
+//	v.Add("tls", "rds")
+//	b := auth.NewConnectionBuilder(endpoint, region, user, dbname, creds)
+//	connectStr, err := b.WithParams(v).WithTCPFormat().Build(context.TODO())
+func (b ConnectionStringBuilder) WithParams(params url.Values) ConnectionStringBuilder {
+	b.params = params
+	return b
+}
+
+// WithFormat will return a builder with the given connection format
+func (b ConnectionStringBuilder) WithFormat(f ConnectionFormat) ConnectionStringBuilder {
+	b.connectFormat = f
+	return b
+}
+
+// WithTCPFormat will set the format to TCP and return the modified builder
+func (b ConnectionStringBuilder) WithTCPFormat() ConnectionStringBuilder {
+	return b.WithFormat(TCPFormat)
+}
+
+// Build will return a new connection string that can be used to open a connection
+// to the desired database.
+//
+//	Example:
+//	b := auth.NewConnectionStringBuilder(endpoint, region, user, dbname, creds)
+//	connectStr, err := b.WithTCPFormat().Build(context.TODO())
+//	if err != nil {
+//		panic(err)
+//	}
+//	const dbType = "mysql"
+//	db, err := sql.Open(dbType, connectStr)
+func (b ConnectionStringBuilder) Build(ctx context.Context) (string, error) {
+	if b.connectFormat == NoConnectionFormat {
+		return "", fmt.Errorf("no connection format was specified")
+	}
+
+	authToken, err := BuildAuthToken(ctx, b.endpoint, b.region, b.user, b.creds)
+	if err != nil {
+		return "", err
+	}
+
+	connectionStr := fmt.Sprintf("%s:%s@%s(%s)/%s",
+		b.user, authToken, string(b.connectFormat), b.endpoint, b.dbName,
+	)
+
+	if len(b.params) > 0 {
+		connectionStr = fmt.Sprintf("%s?%s", connectionStr, b.params.Encode())
+	}
+	return connectionStr, nil
```

Comment:
Nit: can be shortened

```
if len(b.params) > 0 {
    return fmt.Sprintf("%s?%s", connectionStr, b.params.Encode()), nil
}

return fmt.Sprintf("%s:%s@%s(%s)/%s", 
    b.user, authToken, string(b.connectFormat), b.endpoint, b.dbName), nil
```

Reply:
That would not be valid, as that results in a different connection string.